browser/importer/brave_profile_writer.cc

```diff
@@ -441,3 +447,72 @@ void BraveProfileWriter::UpdateWindows(
   if (active)
     active->window()->Show();
 }
+
+
+// NOTE: the strings used as keys match the values found in Muon:
+// browser-laptop/js/data/searchProviders.js
+std::map<std::string,TemplateURLPrepopulateData::PrepopulatedEngine>
+    importable_engines = {
+      {"Amazon", TemplateURLPrepopulateData::amazon},
+      {"Bing", TemplateURLPrepopulateData::bing},
+      {"DuckDuckGo", TemplateURLPrepopulateData::duckduckgo},
+      {"Ecosia", TemplateURLPrepopulateData::ecosia},
+      {"GitHub", TemplateURLPrepopulateData::github},
+      {"Google", TemplateURLPrepopulateData::google},
+      {"Infogalactic", TemplateURLPrepopulateData::infogalactic},
+      {"MDN Web Docs", TemplateURLPrepopulateData::mdnwebdocs},
+      {"Qwant", TemplateURLPrepopulateData::qwant},
+      {"searx", TemplateURLPrepopulateData::searx},
+      {"Semantic Scholar", TemplateURLPrepopulateData::semanticscholar},
+      {"Stack Overflow", TemplateURLPrepopulateData::stackoverflow},
+      {"StartPage", TemplateURLPrepopulateData::startpage},
+      {"Twitter", TemplateURLPrepopulateData::twitter},
+      {"Wikipedia", TemplateURLPrepopulateData::wikipedia},
+      {"Wolfram Alpha", TemplateURLPrepopulateData::wolframalpha},
+      {"Yahoo", TemplateURLPrepopulateData::yahoo},
+      {"Yandex", TemplateURLPrepopulateData::yandex},
+      {"YouTube", TemplateURLPrepopulateData::youtube}
+    };
+
+void BraveProfileWriter::UpdateSettings(const SessionStoreSettings& settings) {
```

Comment:
can you leverage `ProfileWriter::AddKeywords` for importing regular profile search engine?

Reply:
I don't think we can use that; it seems that's for adding new engines altogether, versus choosing an already existing one. It also doesn't seem to set a default